### DISCUSSION TITLE
Unable to render this definition

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#9824] [Questions]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

Hello,

I'm trying to use FastAPI's Swagger UI in an offline environment. I have downloaded the static files from Swagger UI's GitHub repository and put them in a `static` directory in my project. I have also configured FastAPI to serve these static files and to use the local Swagger UI for the docs page.

Here is a simplified version of my code:

```python
import uvicorn
from pathlib import Path
from fastapi import FastAPI
from file_parse.api import fp_router
from fastapi.staticfiles import StaticFiles

ROOT_PATH = Path(__file__).parent

app = FastAPI()

app.mount("/static", StaticFiles(directory="static"), name="static")



app.include_router(fp_router, prefix="/file_parse", tags=["图片、音频、视频解析"])

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8080, log_level="debug", reload=True)
```
However, when I visit the /docs page, I get the following error:
```
Unable to render this definition
The provided definition does not specify a valid version field.

Please indicate a valid Swagger or OpenAPI version field. Supported version fields are swagger: "2.0" and those that match openapi: 3.0.n (for example, openapi: 3.0.0).
```
I have checked my OpenAPI JSON at /openapi.json and it seems to be fine. I'm also using a version of Swagger UI that supports OpenAPI 3.0.0.

I'm not sure what I'm doing wrong. Could you please help me?

fastapi==0.99.1
pydantic==1.10.11
swaggerui v5.1.0

Thank you.

**Comment by hhartzer** [comment]
Not sure why this was converted to a discussion. I'm seeing this as well both on 0.99.1 and 0.100.0-beta3.

**Reply by hhartzer** [reply]
Also, the docs seem to be pinned to Swagger v4. Is that correct?

https://fastapi.tiangolo.com/advanced/extending-openapi/?h=openapi#download-the-files

**Reply by jgould22** [reply]
Swagger ui 4.0 does not support the open api spec 3.1.0 which is what FastAPI > 99 is generating. 

You must update your swagger ui to version 5

**Reply by hhartzer** [reply]
Docs should have links to Swagger v5 in that case. I'll give v5 a try.

**Reply by hhartzer** [reply]
v5 works for me! @sanmaomashi if you really are using Swagger v5, maybe it's in your browser cache and you need to do something like a ctrl+shift+r?

**Comment by jgould22** [comment]
You can get the latest swagger ui from here https://github.com/swagger-api/swagger-ui/releases you need 5.0.0 or greater to support the api definition 3.1.0 that fast API is generating.

**Comment by tiangolo** [comment]
Please provide an example that I can copy paste and run here without modifications to replicate your problem.

As this was created originally as an issue (it should have been a question), it doesn't have the proper template with all the information I/we would need to be able to help you. It's probably better if you create a new discussion question and fill out the entire template, checking each step.

Given that, I'm gonna close this one and wait for the new one with all the data. ☕

**Reply by hhartzer** [reply]
Looks like no bug, but the docs say to download the v4 Swagger instead of the needed v5 Swagger.

Thanks for checking!

**Reply by yxlwfds** [reply]
这解决了我的问题
